src/main/java/com/renatusnetwork/momentum/data/squads/SquadsManager.java

```diff
@@ -0,0 +1,187 @@
+package com.renatusnetwork.momentum.data.squads;
+
+import com.renatusnetwork.momentum.Momentum;
+import com.renatusnetwork.momentum.data.stats.PlayerStats;
+import com.renatusnetwork.momentum.utils.Utils;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.jetbrains.annotations.NotNull;
+
+import java.time.Instant;
+import java.util.*;
+
+public class SquadsManager {
+	private final Set<PlayerStats> inSquadChat;
+	private final Set<PlayerStats> inSquadChatSpy;
+	private final Map<String, Squad> offlineCache; // squad members who go offline without leaving will be stored here for 1 minute
+
+	public SquadsManager() {
+		this.inSquadChat = new HashSet<>();
+		this.inSquadChatSpy = new HashSet<>();
+		this.offlineCache = new HashMap<>();
+	}
+
+	public void createSquad(PlayerStats leader) {
+		Squad squad = Squad.Builder.create().setLeader(leader).build();
+		leader.updateSquad(squad);
+	}
+
+	public void invite(PlayerStats inviter, PlayerStats invitee) {
+		Squad squad = inviter.getSquad();
+		squad.addInvite(invitee);
+
+		new BukkitRunnable() {
+			@Override
+			public void run() {
+				if (squad.hasInvite(invitee)) {
+					squad.removeInvite(invitee);
+					inviter.sendMessage(Utils.translate("&9" + invitee.getName() + " &3did not accept the squad invite in time"));
+					invitee.sendMessage(Utils.translate("&3You did not accept the squad invite in time"));
+					if (squad.size() == 1)
+						disband(squad);
+				}
+			}
+		}.runTaskLater(Momentum.getPlugin(), 20 * 30); // 30 seconds to accept invite
+	}
+
+	public void join(@NotNull Squad squad, PlayerStats newMember) {
+		squad.removeInvite(newMember); // remove invite after joining
+		squad.addMember(newMember);
+		newMember.updateSquad(squad);
+	}
+
+	public void leave(PlayerStats member) {
+		Squad squad = member.getSquad();
+		inSquadChat.remove(member);
+		squad.removeMember(member);
+		member.updateSquad(null);
```

Comment:
Can we add a member.resetSquad() function that sets it to null in the object? A little cleaner than null parameters